src/colorManipulation.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * 
+ * @param blurFrame This is the dimensions of the portion between the two quadrants
+ * @param kernel_size This will be a number that represents ONE side (since its a square)
+ * @returns new ImageData with the same width and height of the blurFrame
+ */
+export function regionConvolution(blurFrame: ImageData, kernel_size: number) {
+    const frameCopy = new Uint8ClampedArray(blurFrame.data.length);
+    for(let row = 0; row < blurFrame.height; row++) { 
+        for(let col = 0; col < blurFrame.width; col++) {
+            let currentPixel = (row * blurFrame.width  + col) * 4;
+            let red = 0, blue = 0, green = 0, alpha = 255;
+            let layers = kernel_size % 3; // represents how many times you "wrap" currentPixel; i.e. 3x3 kernel wraps once, 5x5 wraps twice, etc.
```

Comment:
i like the ingenuity of calculating how many "wraps" there are around the center pixel, but it does add some read complexity. i think a similar, but easier to read, heuristic could be the size of the kernel / 2. what do you think of that?